apps/meteor/app/livechat/server/hooks/saveAnalyticsData.ts

```diff
@@ -70,8 +70,12 @@ callbacks.add(
 			message = { ...(await normalizeMessageFileUpload(message)), ...{ _updatedAt: message._updatedAt } };
 		}
 
-		const analyticsData = getAnalyticsData(room, new Date());
-		await LivechatRooms.getAnalyticsUpdateQueryByRoomId(room, message, analyticsData, roomUpdater);
+		if (isMessageFromVisitor(message)) {
+			await LivechatRooms.getAnalyticsUpdateQueryBySentByVisitor(room, message, roomUpdater);
+		} else {
+			const analyticsData = getAnalyticsData(room, new Date());
+			await LivechatRooms.getAnalyticsUpdateQueryBySentByAgent(room, message, analyticsData, roomUpdater);
+		}
```

Comment:
(this is a personal pref so feel free to resolve if you think it doesn't make sense for this case)

Reply:
Hmmm that was exactly what I tried in my first approach. Gazzo suggested changing it to the current approach since we have two clear distinct paths for each user "type", which I also feel like it's easier to read

Reply:
Fair enough